library/certificate_request.py

```diff
@@ -61,6 +61,31 @@
         if I(name) is not an absolute path.
     required: false
     default: /etc/pki/tls
+  key_usage:
+    description:
+      - Allowed Key Usage for the certificate.
+    choices:
+      - digitalSignature
+      - nonRepudiation
+      - keyEncipherment
+      - dataEncipherment
+      - keyAgreement
+      - keyCertSign
+      - cRLSign
+      - encipherOnly
+      - decipherOnly
+    required: false
+    default:
+      - digitalSignature
+      - keyEncipherment
+  extended_key_usage:
+    description:
+      - Extended Key Usage attributes to be present in the
+        certificate request.
+    required: false
```

Comment:
did you want to list the `choices` here too?  Or can this field take arbitrary values other than the ones listed in `EXTENDED_KEY_USAGE_CHOICES`?

Reply:
Initially I've added `EXTENDED_KEY_USAGE_CHOICES` but then I remembered that during the API design @rcritten explained me that EKU can be any OID so we shouldn't limit it to a small list of known aliases.

I removed the constant.